File: app/src/main/java/batterymonitor/dev/com/batterchargemonitor/DateHelper.java
package batterymonitor.dev.com.batterchargemonitor;

import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.*;

public class DateHelper {

    public static Date getDateWithServerTimeStamp(String str) {

        SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss", Locale.getDefault());
        dateFormat.setTimeZone(TimeZone.getDefault());

        try {
            return dateFormat.parse(str);
        } catch (ParseException var4) {
            return null;
        }
    }

    public static String getDateStr( Date date) {
        SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss", Locale.getDefault());
        dateFormat.setTimeZone(TimeZone.getDefault());

        return dateFormat.format(date);
    }


}

File: app/src/main/java/batterymonitor/dev/com/batterchargemonitor/DateHelper.kt
package batterymonitor.dev.com.batterchargemonitor
import java.text.ParseException
import java.text.SimpleDateFormat
import java.util.*

object DateHelper {

    fun getDateWithServerTimeStamp(str: String): Date? {
        val dateFormat = SimpleDateFormat(
            "yyyy-MM-dd HH:mm:ss",
            Locale.getDefault()
        )
        dateFormat.timeZone = TimeZone.getDefault() // IMP !!!
        try {
            return dateFormat.parse(str)
        } catch (e: ParseException) {
            return null
        }
    }

    fun getDateStr(date: Date): String? {
        val dateFormat = SimpleDateFormat(
            "yyyy-MM-dd HH:mm:ss",
            Locale.getDefault()
        )
        dateFormat.timeZone = TimeZone.getDefault() // IMP !!!
        try {
            return dateFormat.format(date)
        } catch (e: ParseException) {
            return null
        }
    }
}

